### DISCUSSION TITLE
Codespaces IP range?

### DISCUSSION THREAD

**Discussion by mellson** [community/community#6548] [Codespaces] [answered]
I need to connect to a database from a codespace. The database is running at AWS RDS so I need to open the in- and outbound connections to the devices that can connect.

Where do I find the IP addresses used by codespaces? It's not available in the Meta API at the moment.

Or is there another way to solve this?
Can I make the codespace use the IP address of the host device when I run it locally?
Because my local devices are already whitelisted.

**Comment by asciimike** [comment, marked as answer]
> Where do I find the IP addresses used by codespaces? It's not available in the Meta API at the moment.

Currently we don't have them published, and even if we did, you probably wouldn't want to allow all of them access to your database 😅 

An ideal solution would be for us to offer the ability to connect to private networks on major cloud providers; it's something we're investigating, but unfortunately we don't have a solution for you today 😞 

I think the easiest solution might be to use a VPN like Tailscale and add both your codespace and the database to the same "tailnet", at which point you'd be allowed to connect directly to it in a secure manner. https://tailscale.com/kb/1160/github-codespaces/ and https://tailscale.com/kb/1021/install-aws/ might be helpful here.

**Reply by mellson** [reply]
That could indeed be an excellent interim solution. But ideally, as you say, it would be great if you could establish private connections with the major cloud providers. Thanks very much for the reply!
